woocommerce/woocommerce
# [Experimental] Surface custom fields in Emails

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:

This PR is based on https://github.com/woocommerce/woocommerce/pull/43449 which handles additional fields on the order confirmation page. 

This PR adds additional checkout fields to order confirmation emails. Some additional hooks were needed in the template files to support this change.

Closes #42775

HTML version:

![Screenshot 2024-01-11 at 16 17 25](https://github.com/woocommerce/woocommerce/assets/90977/bd16a7a4-6045-4acc-9d3e-997b4ebdd437)

Plain text version:

![Screenshot 2024-01-11 at 16 17 33](https://github.com/woocommerce/woocommerce/assets/90977/1fce6368-cf77-48bc-b374-e9f1f41996c6)



### How to test the changes in this Pull Request:

1. Before testing, implement the snippet to add custom fields as found in https://github.com/woocommerce/woocommerce/pull/43449
2. Go to WC > Settings > Emails > Customer invoice / Order details and enable HTML or Plain text depending on what you're testing. 
3. Add something to the cart and place an order via block based checkout whilst also filling out the new additional fields you've registered via the snippet.
4. In the order received email, confirm you see any additional address data, and additional field data.



### Changelog entry




<details>

#### Significance




#### Type




#### Message 

#### Comment 

This functionality is behind a feature flag.

</details>
